python/bifrost/units.py

```diff
@@ -31,13 +31,16 @@
 ureg = pint.UnitRegistry()
 
 def convert_units(value, old_units, new_units):
-	old_quantity = value * ureg.parse_expression(old_units)
-	try:
-		new_quantity = old_quantity.to(new_units)
-	except pint.DimensionalityError:
-		raise ValueError("Cannot convert units %s to %s" %
+    old_quantity = value * ureg.parse_expression(old_units)
+    try:
+        new_quantity = old_quantity.to(new_units)
+    except pint.DimensionalityError:
+        raise ValueError("Cannot convert units %s to %s" %
 		                 (old_units, new_units))
-	return new_quantity.magnitude
+    if value >= 0:
+        return new_quantity.magnitude
+    else:
+        return -1 * new_quantity.magnitude
```

Comment:
That may indeed be where it comes from. Here's an example pipeline to show where it's popping up:
```python
with bfp.Pipeline() as pipeline:
    
    n_blocks = 128              # Number of blocks in raw file
    n_coarse = 64
    n_t_per_coarse = 524288
    n_pol    = 4                
    n_int    = 128              # Runtime - Number of integrations
    n_chunks = 2                # Runtime - Number of time chunks 
    n_t      = n_blocks / n_int
    n_chan   = n_coarse * n_t_per_coarse * n_chunks
    
    filelist = sorted(glob.glob('*.raw'))[:2]
    
    # Read from guppi raw file
    b_guppi   = blocks.read_guppi_raw(filelist, core=1, buffer_nframe=4)
    b_gup2    = views.rename_axis(b_guppi, 'freq', 'channel')
    # Buffer up two blocks & reshape to allow longer FFT
    with bf.block_scope(fuse=True, core=2):
        b_gup2    = views.split_axis(b_gup2, axis='time', n=n_chunks, label='time_chunk')
        b_gup2    = blocks.transpose(b_gup2, axes=['time', 'channel', 'time_chunk', 'fine_time', 'pol'], buffer_nframe=1)
        b_gup2    = views.merge_axes(b_gup2, 'time_chunk', 'fine_time', label='fine_time')
    
    # Copy over to GPU and FFT
    with bf.block_scope(fuse=True, core=3):
        b_copy    = blocks.copy(b_gup2,  space='cuda')
        b_fft     = blocks.fft(b_copy,  axes='fine_time', axis_labels='freq')
        b_ffs     = blocks.fftshift(b_fft, axes='freq')
        b_pow     = blocks.detect(b_ffs, mode='stokes')

    # Copy back to host and flatten channel/freq axis to form output spectra
    b_copp    = blocks.copy(b_pow, space='system', buffer_nframe=1, core=4)
    #
    b_flat    = views.merge_axes(b_copp, 'channel', 'freq', label='freq')
    blocks.print_header(b_flat)
    pipeline.run()
```

And the exception:

```
Exception in thread Pipeline_0/PrintHeaderBlock_0:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "build/bdist.linux-x86_64/egg/bifrost/pipeline.py", line 316, in run
    self.main(active_orings)
  File "build/bdist.linux-x86_64/egg/bifrost/pipeline.py", line 453, in main
    self.sequence_proclogs[i].update(iseq.header)
  File "build/bdist.linux-x86_64/egg/bifrost/ring2.py", line 288, in header
    hdr = self.header_transform(deepcopy(hdr))
  File "build/bdist.linux-x86_64/egg/bifrost/views/basic_views.py", line 142, in header_transform
    "%f != %f" % (scale1, n*scale2))
ValueError: Scales of merge axes do not line up: -2.929688 != 2.929688
```